src/ripple/app/paths/tests/Path_test.cpp

```diff
@@ -1,212 +1,585 @@
-        expectEquals(srcAmount[jss::currency], "USD");
-        expectEquals(srcAmount[jss::value], "5");
-        expectEquals(srcAmount[jss::issuer], alice.pk.humanAccountID());
+    void
+    test_payment_auto_path_find()
+    {
+        testcase("payment auto path find");
+        Env env(*this);
+        auto const gw = Account("gateway");
+        auto const USD = gw["USD"];
+        env.fund(XRP(10000), "alice", "bob", gw);
+        env.trust(USD(600), "alice");
+        env.trust(USD(700), "bob");
+        env(pay(gw, "alice", USD(70)));
+        env(pay("alice", "bob", USD(24)));
+        env.require(balance("alice", USD(46)));
+        env.require(balance("bob", USD(24)));
     }
 
     void
-    test_payment_auto_path_find_using_build_path()
+    test_path_find()
     {
-        testcase("payment auto path find (using build_path)");
+        testcase("path find");
+        Env env(*this);
+        auto const gw = Account("gateway");
+        auto const USD = gw["USD"];
+        env.fund(XRP(10000), "alice", "bob", gw);
+        env.trust(USD(600), "alice");
+        env.trust(USD(700), "bob");
+        env(pay(gw, "alice", USD(70)));
+        env(pay(gw, "bob", USD(50)));
+
+        auto const alternatives = findPath(env.ledger, "alice", "bob",
+            {USD}, Account("bob")["USD"](5));
+        Json::Value jv;
+        Json::Reader().parse(R"([{
+		        "paths_canonical" : [],
+		        "paths_computed" : [],
+		        "source_amount" :
+		        {
+			        "currency" : "USD",
+			        "issuer" : "r9QxhA9RghPZBbUchA9HkrmLKaWvkLXU29",
+			        "value" : "5"
+		        }
+            }])", jv);
+        expect(jv == alternatives);
+    }
 
-        std::uint64_t const xrp = std::mega::num;
+    void
+    test_path_find_cosume_all()
+    {
+        testcase("path find consume all");
+        Env env(*this);
+        auto const gw = Account("gateway");
+        auto const USD = gw["USD"];
+        env.fund(XRP(10000), "alice", "bob", gw);
+        env.trust(Account("alice")["USD"](600), gw);
+        env.trust(USD(700), "bob");
+
+        auto const alternatives = findPath(env.ledger, "alice", "bob",
+            {USD}, Account("bob")["USD"](1));
+        Json::Value jv;
+        Json::Reader().parse(R"([{
+		        "paths_canonical" : [],
+		        "paths_computed" : [],
+		        "source_amount" :
+		        {
+			        "currency" : "USD",
+			        "issuer" : "r9QxhA9RghPZBbUchA9HkrmLKaWvkLXU29",
+			        "value" : "1"
+		        }
+            }])", jv);
+        expect(jv == alternatives);
+    }
 
-        auto master = createAccount("masterpassphrase", KeyType::ed25519);
+    void
+    test_alternative_path_consume_both()
+    {
+        testcase("alternative path consume both");
+        Env env(*this);
+        auto const gw = Account("gateway");
+        auto const USD = gw["USD"];
+        auto const gw2 = Account("gateway2");
+        auto const gw2_USD = gw2["USD"];
```

Comment:
The names for these gateways and their currencies could be better. I get that you're avoiding the silliness of using real names as in the tests, but I'm finding it too easy to mix up these names, especially remembering that `USD` is associated only with `gw`.